### PR TITLE
Update _space-margin.scss

### PR DESCRIPTION
**Pull request by thekingofspain**
Added .-margin classes

This PR is for...

[Sample page](url)

### What was happening

-

### What was done

-

### Test Steps

1.

### Screenshots

(prefer animated gif)

### Checklist

-   [ ] tested locally
-   [ ] documented the code
-   [ ] clean all warnings and errors of eslint
-   [ ] requires changes in OutSystems (if so, provide a module with changes)
-   [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

**Comment by gnbm** [comment]
Hello [thekingofspain](https://github.com/thekingofspain) :wave:
Thanks for your Pull Request. It looks like this may be your first contribution to an OutSystems Open Source project. Before we look into your contribution, we need you to accept the :memo: [OutSystems Contributors License Agreement](https://outsystems-public-dl.s3-eu-west-1.amazonaws.com/opensource/cla/OutSystems-Contributor-License-Agreement-2020-03-23.pdf).
After you read and accept the terms of the agreement, reply to this message saying `I read, understood and accept the OutSystems CLA`.

After this step, we'll review your Pull Request and provide any feedback if needed. 
Thank you and feel free to keep contributing! 🚀

**Comment by gnbm** [comment]
Hello [thekingofspain](https://github.com/thekingofspain) :wave:
Thanks for your Pull Request. It looks like this may be your first contribution to an OutSystems Open Source project. Before we look into your contribution, we need you to accept the :memo: [OutSystems Contributors License Agreement](https://outsystems-public-dl.s3-eu-west-1.amazonaws.com/opensource/cla/OutSystems-Contributor-License-Agreement-2020-03-23.pdf).
After you read and accept the terms of the agreement, reply to this message saying `I read, understood and accept the OutSystems CLA`.

After this step, we'll review your Pull Request and provide any feedback if needed. 
Thank you and feel free to keep contributing! 🚀

**Comment by gnbm** [comment]
Hello [thekingofspain](https://github.com/thekingofspain)
Before we move forward we need you to answer the previous comment about the OutSystems CLA.
Cheers

File: src/scss/05-useful/_space-margin.scss [hunk 1]
```diff
@@ -30,6 +30,34 @@
 	&-auto {
 		margin: 0 auto;
 	}
+	
+}
+
+///
+.-margin {
```
[the remark this answers is not in the document]
Comment (thekingofspain) [reply src/scss/05-useful/_space-margin.scss:37]: My CSS suggestions are all based around flex box layout.  When working around built in OutSystems margins and using flexbox, applying a negative margins fix some of the sizing /alignment issues.

https://www.rawkblog.com/2019/01/flexbox-gutters-negative-margins-solved/

I tend to use css classes more than custom css in reactive as a way to workaround all the dom element injection that widgets create; thus my rational for negative margin classes.  The custom css that i do write needs these css classes placeholders to pull off something like a CSS4 :has selector.

.display-flex > .OSBlockWidget { display: inherit;}
.justify-content-* > .OSBlockWidget { justify-content: *'}
.full-width > .OSBlockWidget { width: inherit; }

[the remark this answers is not in the document]
Comment (joselrio) [reply src/scss/05-useful/_space-margin.scss:37]: Hi @thekingofspain. 
Well, that said, having this on Framework by default do not much sense. On that case you are able to do it at your fork and use your own approach for it.
Nevertheless, we will take your suggestion in order to investigate about how/if we can deal with it without the need of having useful css selectors with negative margins.

Again, thank you for your feedback!

**Pull request closed without merging**